### ISSUE TITLE
Editing the ID of a page modifies the root object ID, not the page ID

### ISSUE THREAD

**Issue by thecodewarrior** [jgraph/drawio#3397]
### Preflight Checklist

* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Describe the bug**
You can edit the ID of an element by double-clicking the ID in the "Edit Data" dialog (#1125), however when editing the ID of a page it modifies the ID of the root object, not the page itself. The reason I want to edit the page ID is because I'm embedding diagrams in a wiki and I want to be able to select a specific page to render using a URL fragment and the `pageId` key in the `data-mxgraph` JSON, and `/docs/foo.drawio#authentication-process` is nicer than `/foo.drawio#Sehw787GgRkPoYXc5xnK`

**To Reproduce**
1. Create a new diagram
2. Click "File > Properties" and uncheck the "Compressed" checkbox
3. View the diagram file in a text editor. e.g. 
```xml
<mxfile host="Electron" modified="2023-03-02T19:07:04.671Z" agent="5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) draw.io/20.8.10 Chrome/106.0.5249.199 Electron/21.3.5 Safari/537.36" etag="KbgwiEg0Q1go943MyiKs" compressed="false" version="20.8.10" type="device">
  <diagram name="Page-1" id="Sehw787GgRkPoYXc5xnK">
    <mxGraphModel dx="557" dy="378" grid="1" gridSize="10" guides="1" tooltips="1" connect="1" arrows="1" fold="1" page="0" pageScale="1" pageWidth="850" pageHeight="1100" math="0" shadow="0">
      <root>
        <mxCell id="0" />
        <mxCell id="1" parent="0" />
      </root>
    </mxGraphModel>
  </diagram>
</mxfile>
```
4. Go back to draw.io and click "Edit Data" in the format panel
5. Double-click the ID, modify it, and click "Apply" on both dialogs
6. Click "Edit Data" and see that the ID appears to be unchanged
7. Open the diagram in a text editor and note that the ID of the root object has been modified, not the page (`<diagram>` element). e.g.
```xml
<mxfile host="Electron" modified="2023-03-02T19:26:00.013Z" agent="5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) draw.io/20.8.10 Chrome/106.0.5249.199 Electron/21.3.5 Safari/537.36" etag="GWX4WgRpXjUT00hXuiFL" compressed="false" version="20.8.10" type="device">
  <diagram name="Page-1" id="Sehw787GgRkPoYXc5xnK">
    <mxGraphModel dx="1026" dy="743" grid="1" gridSize="10" guides="1" tooltips="1" connect="1" arrows="1" fold="1" page="0" pageScale="1" pageWidth="850" pageHeight="1100" math="0" shadow="0">
      <root>
        <object label="" id="foo-page">
          <mxCell />
        </object>
        <mxCell id="1" parent="foo-page" />
      </root>
    </mxGraphModel>
  </diagram>
</mxfile>
```

**Expected behavior**
I would expect the page ID to be modified, not the root object. e.g.
```xml
<mxfile host="Electron" modified="2023-03-02T19:29:07.148Z" agent="5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) draw.io/20.8.10 Chrome/106.0.5249.199 Electron/21.3.5 Safari/537.36" etag="L_mChIFcVw12wgg-rapE" compressed="false" version="20.8.10" type="device">
  <diagram name="Page-1" id="foo-page">
    <mxGraphModel dx="1026" dy="743" grid="1" gridSize="10" guides="1" tooltips="1" connect="1" arrows="1" fold="1" page="0" pageScale="1" pageWidth="850" pageHeight="1100" math="0" shadow="0">
      <root>
        <mxCell id="0" />
        <mxCell id="1" parent="0" />
      </root>
    </mxGraphModel>
  </diagram>
</mxfile>
```

**Screenshots**

https://user-images.githubusercontent.com/5467669/222534727-c236217d-01d0-4fa3-bc5c-e8057d53a4a3.mp4

**draw.io version (In the Help->About menu of the draw.io editor):**

- 20.8.10

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: N/A
 - Version: macOS Ventura (13.2.1)

**Comment by alderg** [comment]
Thanks for the report. This will be fixed in the next release.

**Comment by davidjgraph** [comment]
Released in 21.1.0
